src/openms/source/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.cpp

```diff
@@ -1532,12 +1556,16 @@ namespace OpenMS
   {
     std::vector<PeakGroup> filtered_pg_vec;
     filtered_pg_vec.reserve(dspec.size());
-
+    //dspec.sort();
     for (Size i = 0; i < dspec.size(); i++)
     {
-      if (i > 0)
+      if (!dspec[i].isTargeted())
       {
-        if (!dspec[i].isTargeted() && abs(dspec[i - 1].getMonoMass() - dspec[i].getMonoMass()) < 1e-3 && dspec[i - 1].getSNR() >= dspec[i].getSNR())
+        if (i>0 && abs(dspec[i - 1].getMonoMass() - dspec[i].getMonoMass()) < 1e-3 && dspec[i - 1].getSNR() >= dspec[i].getSNR())
+        {
+          continue;
+        }
+        if (i< dspec.size() - 1 && abs(dspec[i + 1].getMonoMass() - dspec[i].getMonoMass()) < 1e-3 && dspec[i + 1].getSNR() > dspec[i].getSNR()) // should not >=
```

Comment:
is this safe? e.g. can dspec.size() be 0? then this would produce an underflow

Reply:
Reimplemented in a much simpler algorithm that works better..